.github/workflows/docs.yml

```diff
@@ -0,0 +1,41 @@
+name: check-docs
+
+on:
+  pull_request:
+  # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request
+    types: [opened, reopened, converted_to_draft, ready_for_review, synchronize]
+    branches:
+      - master
+  push:
+    branches:
+    # https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-only-when-a-push-to-specific-branches-occurs
+        - master
+
+jobs:
+  broken-links:
+    strategy:
+      matrix:
+          # Add versions here to expand support matrix
+        python-version: ["3.12"]
```

Comment:
It's fine if we only need to do 3.12 here, but just checking to make sure this shouldn't be expanded to all versions of Python that we support.

Reply:
I don't think `make linkcheck` will work differently for different versions of python. Which is why I chose to have 1 job for this.